src/Numerics/DGMethods/DGModel.jl

```diff
@@ -931,28 +943,38 @@ function launch_volume_gradients!(dg, state_prognostic, t; dependencies)
     Qhypervisc_grad, _ = dg.states_higher_order
 
     info = basic_launch_info(dg)
-    workgroup = (info.Nq, info.Nq)
-    ndrange = (info.Nq * info.nrealelem, info.Nq)
+    # We iterate the kernels horizontally, so our
+    # workgroup is determined by (Nqh, Nqh), where Nqh is
+    # the number of quadrature points in a horizontal (x, y) direction
+    # In three-dimensions, we assume the same polynomial orders in both x, y
+    # directions, hence Nq is the same for both
+    workgroup = (info.Nq[1], info.Nq[2])
+    # Total size of the iteration space
+    ndrange = (info.Nq[1] * info.nrealelem, info.Nq[1])
     comp_stream = dependencies
 
     # If the model direction is EveryDirection, we need to perform
     # both horizontal AND vertical kernel calls; otherwise, we only
     # call the kernel corresponding to the model direction `dg.diffusion_direction`
     if dg.diffusion_direction isa EveryDirection ||
        dg.diffusion_direction isa HorizontalDirection
+
+        # We assume N₁ = N₂, so the same polyorder, quadrature, and differentiation
+        # operators are used
+        horizontal_polyorder = info.N[1]
+        horizontal_D = dg.grid.D[1]
```

Comment:
In the comment here it says `same polyorder, quadrature, and differentiation operators are used`, but then we don't set omega, like we do on line 883 above, i.e., 
```
vertical_ω = grid.ω[dim]
```
Was this intentional?

Reply:
Ah, so the quadrature weights are not an argument for `volume_gradients!`, which is why I didn't grab them here. Some of the kernels that do have them as arguments also don't technically use them also, which can be confusing. We can clean this up, thanks for pointing this out!

Reply:
No worries! I know it's still draft and there will be some cleaning. Every question I ask and every answer I get help me understand the code a bit better :) Thank you!